gym/lib/gym/detect_values.rb

```diff
@@ -98,9 +98,12 @@ def self.detect_selected_provisioning_profiles
 
               bundle_identifier = current["PRODUCT_BUNDLE_IDENTIFIER"]
               provisioning_profile_specifier = current["PROVISIONING_PROFILE_SPECIFIER"]
-              next if provisioning_profile_specifier.to_s.length == 0
-
-              provisioning_profile_mapping[bundle_identifier] = provisioning_profile_specifier
+              provisioning_profile_uuid = current["PROVISIONING_PROFILE"]
+              if !provisioning_profile_specifier.nil? && !provisioning_profile_specifier.empty?
```

Comment:
This could be 
```ruby
if provisioning_profile_specifier.to_s.length > 0
```
makes it a lot easier to read and works with `nil` values too 👍